baseapp/middleware.go

```diff
@@ -58,21 +58,32 @@ func NewMetricsHandler(registry metrics.Registry) func(http.Handler) http.Handle
 }
 
 // LogRequest is an hlog access handler that logs request information.
-func LogRequest(r *http.Request, status, size int, elapsed time.Duration) {
+func LogRequest(r *http.Request, status int, size int64, elapsed time.Duration) {
 	hlog.FromRequest(r).Info().
 		Str("method", r.Method).
 		Str("path", r.URL.String()).
 		Str("client_ip", r.RemoteAddr).
 		Int("status", status).
-		Int("size", size).
+		Int64("size", size).
 		Dur("elapsed", elapsed).
 		Str("user_agent", r.UserAgent()).
 		Msg("http_request")
 }
 
 // RecordRequest is an hlog access handler that logs request information and
 // records request metrics.
-func RecordRequest(r *http.Request, status, size int, elapsed time.Duration) {
+func RecordRequest(r *http.Request, status int, size int64, elapsed time.Duration) {
 	LogRequest(r, status, size, elapsed)
 	CountRequest(r, status, size, elapsed)
 }
+
+func AccessHandler(f func(r *http.Request, status int, size int64, duration time.Duration)) func(next http.Handler) http.Handler {
```

Comment:
Is it worth creating a type alias for the access callback signature, maybe `AccessCallback`?